src/aliceVision/depthMap/SemiGlobalMatchingVolume.cpp

```diff
@@ -271,16 +271,15 @@ void SemiGlobalMatchingVolume::SGMoptimizeVolumeStepZ(int rc, int volStepXY, int
         mvsUtils::printfElapsedTime(tall, "SemiGlobalMatchingVolume::SGMoptimizeVolumeStepZ");
 }
 
-StaticVector<IdValue>* SemiGlobalMatchingVolume::getOrigVolumeBestIdValFromVolumeStepZ(int zborder)
+void SemiGlobalMatchingVolume::getOrigVolumeBestIdValFromVolumeStepZ(StaticVector<IdValue>& out_volumeBestIdVal, int zborder)
 {
     long tall = clock();
 
-    StaticVector<IdValue>* volumeBestIdVal = new StaticVector<IdValue>();
-    volumeBestIdVal->reserve(volDimX * volDimY);
-    volumeBestIdVal->resize_with(volDimX * volDimY, IdValue(-1, 1.0f));
+    out_volumeBestIdVal.reserve(volDimX * volDimY);
+    out_volumeBestIdVal.resize_with(volDimX * volDimY, IdValue(-1, 1.0f));
     unsigned char* _volumeStepZPtr = _volumeStepZ->getDataWritable().data();
     int* _volumeBestZPtr = _volumeBestZ->getDataWritable().data();
-    IdValue* volumeBestIdValPtr = volumeBestIdVal->getDataWritable().data();
+    IdValue* out_volumeBestIdValPtr = out_volumeBestIdVal.getDataWritable().data();
```

Comment:
same here, I'd put an assert and if it is the case deal with the nullptr (but i don't know how the funciton works)

Reply:
It's not possible to be null here.